osquery/events/darwin/diskarbitration.cpp

```diff
@@ -191,13 +200,13 @@ std::string DiskArbitrationEventPublisher::extractUdifChecksum(
 
     uint32_t checksum_size;
     dmg_file.seekg(-156L, std::ios::end);
-    dmg_file.read((char *)&checksum_size, sizeof(checksum_size));
+    dmg_file.read((char*)&checksum_size, sizeof(checksum_size));
     // checksum_size is in big endian and we need to byte swap
     checksum_size = CFSwapInt32(checksum_size);
 
     dmg_file.seekg(-152L, std::ios::end); // checksum offset
-    unsigned char *u_buffer = new unsigned char[checksum_size];
-    dmg_file.read((char *)u_buffer, checksum_size);
+    unsigned char* u_buffer = new unsigned char[checksum_size];
+    dmg_file.read((char*)u_buffer, checksum_size);
```

Comment:
I don't know if it's likely, but read can throw an exception and if it does we don't free the memory in u_buffer.